### PR TITLE
feat: Create a release script for releasing java-operator-plugins

### PR DESCRIPTION
**Pull request by jmrodri**

### CODE REVIEW

**Comment by jmrodri** [comment]
### Invalid version specified
```
$ ./release.sh foo
malformed version: "foo"
```

### Not in a release branch
```
$ ./release.sh v0.0.1
branch: "infra-release" is not a release branch, please create a release branch in the form of vX.Y.x
```

### Uncommitted changes
```
$ ./release.sh v0.0.1
directory has uncommitted files
```

### Full release
```
$ ./release.sh v0.0.1
Running tests
Checking for license header...
ok      github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha 0.011s  coverage: 75.0% of statements
?       github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/scaffolds       [no test files]
ok      github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/scaffolds/internal/templates    0.006s  coverage: 29.4% of statements
?       github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/scaffolds/internal/templates/controller [no test files]
?       github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/scaffolds/internal/templates/model      [no test files]
ok      github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/scaffolds/internal/templates/util       0.007s  coverage: 100.0% of statements
ok      github.com/operator-framework/java-operator-plugins/pkg/quarkus/v1alpha/util    0.007s  coverage: 100.0% of statements
Creating a new tag for java-operator-plugins version v0.0.1
warning: refname 'v0.0.1' is ambiguous.
object dc02856add27bf54818dfc1bfc0b4efe631e6a10
type commit
tag v0.0.1
tagger jesus m. rodriguez <jesusr@redhat.com> 1630165665 -0400

java-operator-plugins v0.0.1
gpg: Signature made Sat 28 Aug 2021 11:47:45 AM EDT
gpg:                using DSA key BF6F6F18846753754CBB1DDFBC9679ED89ED8983
gpg: Good signature from "Jesus Miguel Rodriguez (jesusr) <jesusr@redhat.com>" [ultimate]
gpg:                 aka "jesus rodriguez (jmrodri) <jmrodri@gmail.com>" [ultimate]

Done forget to:

git push upstream <release-branch>
git push upstream v0.0.1

Also update the release notes in github for this tag.
```

**Comment by jmrodri** [comment]
Fixes #30

File: release.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+make lint test
+
+# tag the release commit and verify its tag
+echo "Creating a new tag for java-operator-plugins version $VER"
+git tag --sign --message "java-operator-plugins $VER" "$VER"
+git verify-tag --verbose $VER
+
+# remind the releaser to do the next steps
+echo ""
+echo "Done forget to:"
```
Comment (metacosm) [inline release.sh:69]:
```suggestion
echo "Don't forget to:"
```

**Review by metacosm — CHANGES_REQUESTED** [review]

**Comment by metacosm** [comment]
Looks good to me, though, perhaps it would make sense to investigate using GitHub actions for this at some point?

**Comment by jmrodri** [comment]
@metacosm fixed the typo (that's what I get for copying code :) and I am not opposed to the github action. The one we have on operator-sdk is a bit more complicated so I didn't want to use that.

**Review by metacosm — APPROVED** [review]

**Review by asmacdo — APPROVED** [review]

**Review by laxmikantbpandhare — APPROVED** [review]

**Pull request merged**